### DISCUSSION TITLE
Control brightness using proximity sensor

### DISCUSSION THREAD

**Discussion by fkonvick** [seaky/nspanel_pro_tools_apk#35] [Ideas]
I would like to keep the panel at the minimum brightness, but go bright for a limited time triggered by proximity (I tested >20) or tap.  With 1.1, I needed to set the sleep mode to 15/30 secs in the system settings.  This kind of works for the first time, but next time the device goes to sleep (turns the display off completely) rather than just dimming.  Is this something that will work with the new release?  Or is it something that works even with 1.1 and I just overlooked something?

**Comment by seaky** [comment]
If you set System -> Developer mode -> Stay awake to on, the device will never go to sleep. 

Unfortunately  you cant trigger "out dimming" via proximity senors because technically if the screen is on the proximity sensor is not active. Original purpuse was to turn the screen on by proximity. 

But the good thing is out of the box the screen touch will cease the screen dim. (android feature)

**Comment by fkonvick** [comment]
Thanks.  I achieved what I wanted using Automagic.

The idea is that the panel is at, say, 5% brightness all the time, but when the proximity sensor detects someone nearby, it sets the brightness at a high value (like 90%), and keeps it that way until there has been no proximity detected for some time (I use 15 seconds).

Automagic keeps reporting the sensor values all the time, even when the panel is on (and the above works for me anyway), so I'm not sure what you meant.  I think that this would be a common wish, so potentially still a nice feature to implement.

**Reply by seaky** [reply]
Yep the automagic reports even if the screen is on because they implemented. I intentionally turned off proximity listener because as I wrote the original intention was different.

My idea was if you turned on the screen you dont need the proximity sensor. Btw very easy to implement it much harder to put the configuration on the UI. But i wont get away with this I have to implement it in the 2.1 version because the HA integration much more usable with all time sensors.

So stay tuned 2.1 will have more new things.

btw 2.0 is on the way a few days and it will be available.

**Comment by seaky** [comment]
Hi,

I've played a lot with this idea and I have a feasible direction so will be in 2.1

**Comment by fkonvick** [comment]
Cool, thanks :) Anyway, a strong pullback for myself is whether anything I run on my hardware is open sourced.  Are you not considering sharing the code?

**Comment by seaky** [comment]
My App contains lot of reverse-engineered info thereby if I release the source code I make it public. So i dont want to play with jurists.
